packages/components/src/internal/components/administration/GroupManagement.tsx

```diff
@@ -0,0 +1,284 @@
+            const deletedGroupIds = await Promise.all(
+                deletedGroups.map(async groupId => {
+                    const createGroupResponse = await api.security.deleteGroup(parseInt(groupId, 10), projectPath);
+                    return createGroupResponse.deleted;
+                })
+            );
+            deletedGroupIds.forEach(deletedGroup => {
+                delete newGroupMembership[deletedGroup];
+            });
+
+            // Add new members
+            for (const groupId of Object.keys(newGroupMembership)) {
+                const currentMembers = newGroupMembership[groupId].members.map(member => member.id);
+                const oldMembers = new Set(savedGroupMembership[groupId]?.members.map(member => member.id));
+                const addedMembers = currentMembers.filter(id => !oldMembers.has(id));
+                if (addedMembers.length)
+                    await api.security.addGroupMembers(parseInt(groupId, 10), addedMembers, projectPath);
+            }
+
+            // Delete members
+            Object.keys(newGroupMembership).map(async groupId => {
+                const currentMembers = new Set(newGroupMembership[groupId].members.map(member => member.id));
+                const oldMembers = savedGroupMembership[groupId]?.members.map(member => member.id);
+                const deletedMembers = oldMembers?.filter(id => !currentMembers.has(id));
+                if (deletedMembers?.length)
+                    await api.security.removeGroupMembers(parseInt(groupId, 10), deletedMembers, projectPath);
+            });
+
+            // Save updated state
+            setSavedGroupMembership(newGroupMembership);
+            setGroupMembership(newGroupMembership);
+
+            dismissNotifications();
+            createNotification('Successfully updated groups and assignments.');
+        } catch (e) {
+            setErrorMsg(resolveErrorMessage(e) ?? 'Failed to update groups.');
+        }
+
+        setIsDirty(false);
+    }, [
+        setIsDirty,
+        groupMembership,
+        savedGroupMembership,
+        dismissNotifications,
+        createNotification,
+        api.security,
+        projectPath,
+    ]);
+
+    const renderButtons = useCallback(() => {
+        const row = { Modified: { value: lastModified } };
+
+        return (
+            <>
+                <CreatedModified row={row} />
+                <ManageDropdownButton collapsed id="admin-page-manage" pullRight>
+                    <MenuItem href={AppURL.create('audit', 'groupauditevent').toHref()}>View Audit History</MenuItem>
+                </ManageDropdownButton>
+            </>
+        );
+    }, [lastModified]);
+
+    const createGroup = useCallback(
+        (name: string) => {
+            setGroupMembership({ ...groupMembership, [name]: { groupName: name, members: [] } });
+        },
+        [groupMembership]
+    );
```

Comment:
If you use the function form of setGroupMembership you can make  useCallback take no args for the dependency array, which will reduce re-renders.
```suggestion
    const createGroup = useCallback((name: string) => {
        setGroupMembership(current => ({ ...current, [name]: { groupName: name, members: [] } }));
    }, []);
```